libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c

```diff
@@ -96,32 +169,94 @@ int suiteTearDown( int numFailures )
     return numFailures;
 }
 
-/* ======================  TESTING C_Initialize  ============================ */
+/* ==========================  Helper functions  ============================ */
 
 /*!
- * @brief C_Initialize happy path.
+ * @brief Helper function to initialize PKCS #11.
  *
  */
-void test_pkcs11_C_Initialize( void )
+static CK_RV prvInitializePkcs11( SemaphoreHandle_t pxPKCS11Mutex )
 {
     CK_RV xResult = CKR_OK;
 
-    /* Hack: Giving out pointer to xResult when creating a mutex. This unit testcase
-     * will never use the actual mutex. */
-    xQueueCreateMutex_IgnoreAndReturn( ( SemaphoreHandle_t ) &xResult );
+    xQueueCreateMutex_IgnoreAndReturn( pxPKCS11Mutex );
     CRYPTO_Init_Ignore();
     mbedtls_entropy_init_Ignore();
     mbedtls_ctr_drbg_init_Ignore();
     mbedtls_ctr_drbg_seed_IgnoreAndReturn( 0 );
     xResult = C_Initialize( NULL );
 
-    TEST_ASSERT_EQUAL( CKR_OK, xResult );
```

Comment:
Shouldn't the test check every API call?

Reply:
Good point, I will re-add this assert

Reply:
I am leaving it as is. I forgot I was using this helper function for different branches, so this is not guaranteed to be OK. So the unit test case must always check the return of this helper function.